### PR TITLE
Update extend_linter_integration

### PR DESCRIPTION
**Pull request by yash-nisar**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: data/projects/extend_linter_integration.md [hunk 1]
```diff
@@ -35,6 +33,143 @@ to include the newly written Linter bears.
...
+**APPROACH:** I plan to eliminate this issue by integrating this plugin
+(https://www.npmjs.com/package/remark-lint-no-trailing-spaces) which is
+expected to remove trailing whitespaces for all cases. The plugin will be
+installed via package.json and then respectively passed on as shell argument
+to the remark tool.
+
+**ISSUE:** [Markdown parsing issues](https://github.com/coala/coala-bears/issues/940)
+
+**APPROACH:** Likely to be solved by Use `remark-lint` to report meaningful
+messages since all errors will be reported by that commit.
```
Comment (Makman2) [inline data/projects/extend_linter_integration.md]: don't understand this :)

File: data/projects/extend_linter_integration.md [hunk 2]
```diff
@@ -1,14 +1,12 @@
-### Overview
-
+#### OVERVIEW
```
Comment (Makman2) [inline data/projects/extend_linter_integration.md]: why suddenly this change? :D

File: data/projects/extend_linter_integration.md [hunk 3]
```diff
@@ -35,6 +33,143 @@ to include the newly written Linter bears.
...
+
+**APPROACH:** This patch will be vital and aims to use the `remark-lint` plugin
+to report fruitful errors instead of tidying the markdown automatically, because
+in CI mode, the tidying of the file is useless and descriptive messages are
+critical. Since it is already being included by package.json, implementing
+it won’t be a difficult task.
+
+**ISSUE:** [Add remark-validate-links plugin](https://github.com/coala/coala-bears/issues/924)
+
+**APPROACH:** This commit will add a plugin that checks that the local files exist
```
Comment (Makman2) [inline data/projects/extend_linter_integration.md]: Which commit?

File: data/projects/extend_linter_integration.md [hunk 4]
```diff
@@ -35,6 +33,143 @@ to include the newly written Linter bears.
...
+relative to the markdown
+file(https://www.npmjs.com/package/remark-validate-links). The plugin will be
+installed via package.json and then respectively passed on as shell argument
+to the remark tool or in the config file, depending on the requirement.
+
+**ISSUE:** [MarkdownBear table settings](https://github.com/coala/coala-bears/issues/1008)
+
+**APPROACH:** Likely to be solved by Use `remark-lint` to report meaningful
+messages since all errors will be reported by the patch.
```
Comment (Makman2) [inline data/projects/extend_linter_integration.md]: hm copy&paste from above

File: data/projects/extend_linter_integration.md [hunk 5]
```diff
@@ -46,18 +46,18 @@ to include the newly written Linter bears.
 
 * Modification of the testing API to display user friendly results and add
   corresponding documentation.
-* Implementation of the StylintBear.
-* Implementation of the ErlangBear.
+* Implementation of the [StylintBear.](https://github.com/coala/coala-bears/issues/754)
+* Implementation of the [TextLintBear.](https://github.com/coala/coala-bears/issues/1576)
```
Comment (Makman2) [inline data/projects/extend_linter_integration.md]: you want to link the period too? :) looks stylistically a bit better if you do
```
[TextLintBear](...).
```

File: data/projects/extend_linter_integration.md (hunk 2 above)
Comment (yash-nisar) [inline data/projects/extend_linter_integration.md]: Wanted to keep everything in upper case so.., should I revert it ?

Comment (Makman2) [inline data/projects/extend_linter_integration.md]: no it's okay, ideally another commit for that :+1:

File: data/projects/extend_linter_integration.md (hunk 4 above)
Comment (yash-nisar) [inline data/projects/extend_linter_integration.md]: Nope, See https://github.com/coala/coala-bears/issues/1008#issuecomment-261224591 and https://github.com/coala/coala-bears/issues/940#issuecomment-256615428. The [same patch](https://github.com/coala/coala-bears/issues/926) is likely to solve both the issues. :wink:

File: data/projects/extend_linter_integration.md (hunk 3 above)
Comment (yash-nisar) [inline data/projects/extend_linter_integration.md]: Corrected. :+1:

File: data/projects/extend_linter_integration.md (hunk 5 above)
Comment (yash-nisar) [inline data/projects/extend_linter_integration.md]: Yep, looks neat with this^^. :smile:

File: data/projects/extend_linter_integration.md (hunk 1 above)
Comment (yash-nisar) [inline data/projects/extend_linter_integration.md]: Improved the description. :+1:

File: data/projects/extend_linter_integration.md (hunk 4 above)
Comment (Makman2) [inline data/projects/extend_linter_integration.md]: ah okay we used a different executable.

File: data/projects/extend_linter_integration.md [hunk 6]
```diff
@@ -1,28 +1,174 @@
...
+
+**APPROACH:** I plan to eliminate this issue by integrating this plugin
+(https://www.npmjs.com/package/remark-lint-no-trailing-spaces) which is
+expected to remove trailing whitespaces for all cases. The plugin will be
+installed via package.json and then respectively passed on as shell argument
+to the remark tool.
+
+**ISSUE:** [Markdown parsing issues](https://github.com/coala/coala-bears/issues/940)
+
+**APPROACH:** Likely to be solved by Use `remark-lint` to report meaningful
```
Comment (Makman2) [inline data/projects/extend_linter_integration.md]: Use -> using

File: data/projects/extend_linter_integration.md [hunk 7]
```diff
@@ -1,28 +1,174 @@
...
+
+**APPROACH:** This commit will add a plugin that checks that the local files exist
+relative to the markdown
+file(https://www.npmjs.com/package/remark-validate-links). The plugin will be
+installed via package.json and then respectively passed on as shell argument
+to the remark tool or in the config file, depending on the requirement.
+
+**ISSUE:** [MarkdownBear table settings](https://github.com/coala/coala-bears/issues/1008)
+
+**APPROACH:** Likely to be solved by Use `remark-lint` to report meaningful
```
Comment (Makman2) [inline data/projects/extend_linter_integration.md]: Use -> using

**Comment by Makman2** [comment]
ack b0b8a72 949c9d9 5760245

**Comment by Makman2** [comment]
ack ba7fe1b c1c2f86 818aab8 2db88ba

**Comment by Makman2** [comment]
@rultor merge

**Pull request merged**